File: 0260/solution.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int xor = 0;
        for (int num : nums) {
            xor ^= num;
        }
        int bit = 0;
        for (int i = 0; i < 32; i++) {
            if (((1 << i) & xor) != 0) {
                bit = 1 << i;
                break;
            }
        }
        int first = 0;
        for (int num : nums) {
            if ((num & bit) != 0) {
                first ^= num;
            }
        }
        int second = xor ^ first;
        return new int[] { first, second };
    }
}

// time O(n)
// space O(1)

File: 0260/solution.py
class Solution:
    def singleNumber(self, nums: list[int]) -> list[int]:
        xor = 0
        for num in nums:
            xor ^= num
        bit = 0
        for i in range(32):
            if (1 << i) & xor:
                bit = 1 << i
                break
        first = 0
        for num in nums:
            if num & bit:
                first ^= num
        return [first, xor ^ first]
